pandas/core/indexes/period.py

```diff
@@ -460,56 +446,14 @@ def is_full(self):
     days_in_month = _wrap_field_accessor('days_in_month')
     daysinmonth = days_in_month
 
-    @property
-    def start_time(self):
-        return self.to_timestamp(how='start')
-
-    @property
-    def end_time(self):
-        return self.to_timestamp(how='end')
-
```

Comment:
I would leave those properties here, since we will need to add them back when doing the actual refactor

(or you can call the ArrayMixin version like you did for `to_period`/`to_timestamp`, but in this case that will be more verbose than what is there now (but which is not necessarily a problem, we can simplify when doing the actual refactor))

Reply:
Ah, so on this PR, these are inherited from `DatetimelikeArrayMixin`, which won't be a parent of PeriodIndex in the future?

Reply:
I think in the future, assuming we do away with inheritance (which I'm still -0 on), we'll use something akin to 

```
@inherits_from_array(["start_time", "end_time", ...]
class PeriodIndex(...):
```

to succinctly implement appropriate wrapping.

In the interim, removing code from the Index subclasses makes it easier for me to sort out what methods still need to be migrated.

Reply:
> I think in the future, we'll use something akin to

Yes, that might certainly be (or maybe with a method adding them like we do for eg arithmetic methods), but I don't think we will do that in an initial PR that does the refactor, so moving it away now, will make that PR bigger / less easy to understand

> In the interim, removing code from the Index subclasses makes it easier for me to sort out what methods still need to be migrated.

Then I would suggest calling the ArrayMixin implementation as you did for to_timestamp/to_array. That also makes it clear what is already moved and what not yet

Reply:
Sure, will do.

Reply:
Actually given the CI turnaround, if there’s consensus here I’d like to make this change in a follow-up.

Reply:
I would do it here (travis is still fast)

Reply:
done